perceval/utils/logging.py

```diff
@@ -0,0 +1,249 @@
+_CHANNEL_NAMES = ["user", "general", "resources"]
+
+_LOGGING_OFF = python_logging.CRITICAL + 10
+
+_LEVEL_CONVERTER = {
+    python_logging.DEBUG: exqalibur_logging.level.debug,
+    python_logging.INFO: exqalibur_logging.level.info,
+    python_logging.WARNING: exqalibur_logging.level.warn,
+    python_logging.ERROR: exqalibur_logging.level.err,
+    python_logging.CRITICAL: exqalibur_logging.level.critical,
+    _LOGGING_OFF: exqalibur_logging.level.off,
+}
+
+_CHANNEL_CONVERTER = {
+    "general": exqalibur_logging.channel.general,
+    "resources": exqalibur_logging.channel.resources,
+    "user": exqalibur_logging.channel.user,
+}
+
+
+class LoggerConfig(dict):
+    def __init__(self):
+        super().__init__()
+        self.set_to_default_config()
+        self._load_from_persistent_data()
+
+    def set_to_default_config(self):
+        self[_USE_PYTHON_LOGGER] = False
+        self[_ENABLE_FILE] = False
+        self[_CHANNELS] = {}
+        for name in ["general", "resources"]:
+            self._init_channel(name)
+        self._init_channel("user", python_logging.WARNING)
+
+    def _init_channel(self, name: str, level: int = _LOGGING_OFF):
+        if name not in _CHANNEL_NAMES:
+            raise KeyError(f"Unknown channel {name}")
+        if level and level not in _LEVEL_CONVERTER.keys():
+            raise KeyError(f"Unknown level {level} for channel {name}")
+        self[_CHANNELS][name] = {}
+        self[_CHANNELS][name]["level"] = level
+
+    def _load_from_persistent_data(self) -> str:
+        persistent_data = PersistentData()
+        config = persistent_data.load_config()
+        if config and _LOGGING in config:
+            config = config[_LOGGING]
+            if _CHANNELS in config:
+                for key in config[_CHANNELS]:
+                    if key in _CHANNEL_NAMES:
+                        self[_CHANNELS][key] = config[_CHANNELS][key]
+            if _ENABLE_FILE in config:
+                self[_ENABLE_FILE] = config[_ENABLE_FILE]
+
+    def set_level(self, level: int, channel: str = "user"):
+        self[_CHANNELS][channel]["level"] = level
+
+    def save(self):
+        persistent_data = PersistentData()
+        persistent_data.save_config({_LOGGING: dict(self)})
+
+    def message_has_to_be_logged(self, record) -> bool:
+        if "channel" in record.__dict__:
+            if record.levelno < self[_CHANNELS][record.channel]["level"]:
+                return False
+        return True
+
+
+CONFIG = LoggerConfig()
+
+
+class ExqaliburLogger():
+    @staticmethod
+    def initialize():
+        persistent_data = PersistentData()
+        log_path = ExqaliburLogger.get_log_file_path()
+        if persistent_data.is_writable():
+            exqalibur_logging.initialize(log_path)
+        else:
+            exqalibur_logging.initialize()
+
+        if _ENABLE_FILE in CONFIG and CONFIG[_ENABLE_FILE]:
+            print(f"starting to write logs in {log_path}")
```

Comment:
I'd remove this print

Reply:
I'd like to keep it, it's handy to know where the log are.